src/i_winmusic.c

```diff
@@ -1488,28 +1498,25 @@ static void I_WIN_SetMusicVolume(int volume)
 
     volume_factor = sqrtf((float)volume / 15);
 
-    update_volume = (song.file != NULL);
+    if (win_midi_state == STATE_PLAYING)
+    {
+        win_midi_state = STATE_VOLUME;
+    }
 }
 
 static void I_WIN_StopSong(void *handle)
 {
     MMRESULT mmr;
 
-    if (!hPlayerThread)
+    if (!hPlayerThread || !hMidiStream)
     {
         return;
     }
 
-    SetEvent(hExitEvent);
+    win_midi_state = STATE_SHUTDOWN;
```

Comment:
Fixed in https://github.com/fabiangreffrath/woof/pull/1248/commits/62389bba5db1d960378b5ada3c9514db642cf221

Now we wait for the `MOM_DONE` message to stop the player so that no MIDI messages are left in the queue. This can happen quite slowly, depending on the tempo/MIDI file. For example, the DOOM2.WAD MAP07 track stops slowly. But I believe the crashes were due to the player exiting/switching quickly (`hExitEvent`) and the messages getting stuck in the queue (maybe the user has a big delay in VirtualMIDISynth?).